prefect_dbt/cloud/jobs.py

```diff
@@ -319,9 +328,217 @@ def my_flow():
+    run_id: str,
+    run_info: Dict[str, Any],
+):
+    """
+    Compiles a list of steps (commands) to retry, then either build trigger job
+    run options from scratch if it does not exist, else overrides the existing.
+    """
+    steps_override = []
+    for run_step in run_info["run_steps"]:
+        status = run_step["status_humanized"].lower()
+        # Skipping cloning, profile setup, and dbt deps - always the first three
+        # steps in any run, and note, index starts at 1 instead of 0
+        if run_step["index"] <= 3 or status == "success":
+            continue
+        # get dbt build from "Invoke dbt with `dbt build`"
+        command = run_step["name"].partition("`")[2].partition("`")[0]
+
+        # find an executable command like `build` or `run`
+        # search in a list so that there aren't false positives, like
+        # `"run" in "dbt run-operation"`, which is True; we actually want
+        # `"run" in ["dbt", "run-operation"]` which is False
+        command_components = shlex.split(command)
+        for exe_command in EXE_COMMANDS:
+            if exe_command in command_components:
+                break
+        else:
+            exe_command = ""
+
+        is_exe_command = exe_command in EXE_COMMANDS
+        is_not_success = status in ("error", "skipped", "cancelled")
+        is_skipped = status == "skipped"
+        if (not is_exe_command and is_not_success) or (is_exe_command and is_skipped):
+            # if no matches like `run-operation`, we will be rerunning entirely
+            # or if it's one of the expected commands and is skipped
+            steps_override.append(command)
+        else:
+            # errors and failures are when we need to inspect to figure
+            # out the point of failure
+            try:
+                run_artifact_future = await get_dbt_cloud_run_artifact.submit(
+                    dbt_cloud_credentials=dbt_cloud_credentials,
+                    run_id=run_id,
+                    path="run_results.json",
+                    step=run_step["index"],
+                )
+                run_artifact = await run_artifact_future.result()
+            except DbtCloudGetRunArtifactFailed:
+                # get the run results scoped to the step which had an error
+                # an error here indicates that either:
+                # 1) the fail-fast flag was set, in which case
+                #    the run_results.json file was never created; or
+                # 2) there was a problem on dbt Cloud's side saving
+                #    this artifact
+                steps_override.append(command)
+            else:
+                # we only need to find the individual nodes for those run commands
+                run_results = run_artifact["results"]
+                run_nodes = " ".join(
+                    run_result["unique_id"].split(".")[2]
+                    for run_result in run_results
+                    # "fail" here instead of "cancelled"
```

Comment:
Move this somewhere else and also expand on it
"Nope. cancelled is a status that can occur on a run, not an individual node."
```suggestion
                    # "fail" here instead of "cancelled"
```